src/main/java/org/spongepowered/api/text/Texts.java

```diff
@@ -271,53 +383,93 @@ public static Text format(Text template, Map<String, ?> replacements) {
      *
      * @param template The template text in which all {@link Placeholder}s
      *        should be replaced
-     * @param replacements The values available to replace the placeholders. May
-     *        contain null values to skip the placeholder
+     * @param context The context data used to calculate the replacements for
+     *        the {@link Placeholder}s. It does not have to contain enough data
+     *        to replace all placeholders
      * @return The text with all possible placeholders replaced
      */
-    public static Text format(Text template, Object... replacements) {
+    public static Text format(Text template, Map<String, ?> context) {
         checkNotNull(template, "template");
-        checkNotNull(replacements, "values");
-        Map<String, Object> replacementsMap = new HashMap<>();
+        checkNotNull(context, "context values");
+        if (context.isEmpty()) {
+            return template;
+        }
+        return formatNoChecks(template, key -> context.get(key));
+    }
+
+    /**
+     * Creates a new Text instance with all {@link Placeholder}s replaced. All
+     * placeholders with an {@link Optional#empty()} replacement are ignored.
+     * All replacements will be wrapped in a {@link Text} using
+     * {@link Texts#of(Object...)} the {@link TextColor}, {@link TextStyle}s and
+     * {@link TextAction}s from the placeholder are transfered to that method as
+     * well.
+     *
+     * @param template The template text in which all {@link Placeholder}s
+     *        should be replaced
+     * @param context The context data used to calculate the replacements for
+     *        the {@link Placeholder}s. It does not have to contain enough data
+     *        to replace all placeholders
+     * @return The text with all possible placeholders replaced
+     */
+    public static Text format(Text template, Object... context) {
+        checkNotNull(template, "template");
+        checkNotNull(context, "context values");
+        if (context.length == 0) {
+            return template;
+        }
+        final Map<String, Object> replacementsMap = Maps.newHashMapWithExpectedSize(context.length);
         int index = 0;
-        for (Object replacement : replacements) {
+        for (Object replacement : context) {
             replacementsMap.put(Integer.toString(index++), replacement);
         }
-        return formatNoChecks(template, replacementsMap);
+        return formatNoChecks(template, key -> replacementsMap.get(key));
+    }
+
+    /**
+     * Prepares whe given {@link Text} template for the insertion of
+     * replacements using a fluent API.
+     *
+     * @param template The template that should be formatted
+     * @return The format builder used to setup the replacements
+     * @see FormatBuilder
+     */
+    public static FormatBuilder prepare(Text template) {
```

Comment:
Why did you separate the implementation of text formatting into two different methods, format and prepare?